### DISCUSSION TITLE
What should be the values of BGTaskSchedulerPermittedIdentifiers ?

### DISCUSSION THREAD

**Discussion by dhruvgohiloo7** [shinyorg/shiny#877] [Q&A] [answered]
I have implemented one job that has to run when app goes into the background. Do I have to put same values of BGTaskSchedulerPermittedIdentifiers as it is in the [Sample's Info.plist](https://github.com/shinyorg/samples/blob/main/Jobs/Sample.iOS/Info.plist) ? Also do I have to specify those values anywhere else ?

Also I have one task which starts when app is in the foreground and it has to be continued if app goes in to the background, this is how I have implemented it
`ShinyHost.Resolve<IJobManager>().RunTask("Syncing Data", async (c) => await Sync());`
Is this correct ?

**Comment by aritchie** [comment, marked as answer]
Those values only need to be declared in the info.plist.  

Yes - that will work and will continue to run for roughly 30 seconds if your app backgrounds.  Jobs (implementing IJob) are recommended

**Reply by dhruvgohiloo7** [reply]
@aritchie Thank you so much for the help. 
One more question relating to above comment : Creating IJob and run it in the foreground will also continue in the background if app immediately backgrounds and it will also repeat if Repeat value is true after some interval (which decides by iOS) in the background?

**Reply by aritchie** [reply]
Yes

**Reply by dhruvgohiloo7** [reply]
Thank you so much for the confirmation, your time and help. Much appreciated !

**Reply by dhruvgohiloo7** [reply]
@aritchie I have updated shiny to the latest preview version after it I am getting below error

> Shiny.Generators/Shiny.Generators.iOSAppDelegateSourceGenerator/AppDelegate.cs(189,189): Error CS0117: 'Platform' does not contain a definition for 'PerformActionForShortcutItem' (CS0117)

I have added
`[assembly: Shiny.ShinyApplication(
    ShinyStartupTypeName = nameof(MyApp.ShinyAppStartup),
    XamarinFormsAppTypeName = nameof(Myapp)
)]`

**Reply by aritchie** [reply]
If you find an issue, file an issue and send a reproducible sample

**Reply by dhruvgohiloo7** [reply]
@aritchie I have just created new project and It works just fine. But in my current project if I add that assembly([assembly: Shiny.ShinyApplication.....]) then I start getting same error. Any idea why I am getting that error ?

**Reply by aritchie** [reply]
@dhruvgohiloo7 No, I have no idea because you haven't given me a reproducible sample in the proper place as I have requested.
